app/tests/test_collection_model.py

```diff
@@ -51,20 +55,105 @@ def test_create_collection_invalid_providers(self):
+        Factory().create_collection_sample(
             name="collection-required-only",
             sample="collection-1",
             db_create=True,
             required_only=True
         )
+
+    def test_collection_update_on_item_write_operations(self):
+        # assert that collection's updated property is updated when an item is
+        # added to the collection, this item is updated and this item is deleted
+
+        # check collection's update on item insertion
+        initial_last_modified = self.collection.model.updated
+        sleep(0.01)
+        item = Factory().create_item_sample(self.collection.model, sample='item-1', db_create=True)
+        item.model.full_clean()
+        item.model.save()
+        self.collection.model.refresh_from_db()
+        self.assertGreater(
+            self.collection.model.updated,
+            initial_last_modified,
+            msg="Collection's updated property was not updated on item insert"
+        )
+
+        # check collection's update on item update
+        initial_last_modified = self.collection.model.updated
+        sleep(0.01)
+        item.model.name = "new_randon_name"
+        item.model.full_clean()
+        item.model.save()
+        self.collection.model.refresh_from_db()
+        self.assertGreater(
+            self.collection.model.updated,
+            initial_last_modified,
+            msg="Collection's updated property was not updated on item update"
+        )
+
+        # check collection's update on item deletion
+        initial_last_modified = self.collection.model.updated
+        sleep(0.01)
+        item.model.delete()
+        self.collection.model.refresh_from_db()
+        self.assertGreater(
+            self.collection.model.updated,
+            initial_last_modified,
+            msg="Collection's updated property was not updated on item deletion"
+        )
+
+    def test_collection_update_on_asset_write_operations(self):
+        # assert that collection's updated property is updated when an asset is
+        # added to an item of the collection, this asset is updated and this asset is deleted
+
+        # check collection's update on asset insertion
+        item = Factory().create_item_sample(self.collection.model, sample='item-1', db_create=True)
+        initial_last_modified = self.collection.model.updated
+        sleep(0.01)
+        asset = Factory().create_asset_sample(item=item.model, sample='asset-1', db_create=True)
+        asset.model.full_clean()
+        asset.model.save()
+        self.collection.model.refresh_from_db()
+        self.assertGreater(
+            self.collection.model.updated,
+            initial_last_modified,
+            msg="Collection's updated property was not updated on asset insert"
+        )
+
+        # check collection's update on asset update
+        initial_last_modified = self.collection.model.updated
+        sleep(0.01)
+        asset.model.name = f"new-{asset.model.name}"
```

Comment:
Renaming asset is not allowed (see comment about item renaming). Update another attribute instead like the `title`

Reply:
Sure, should have remembered that :see_no_evil: 
Guess the test passes, as the "renaming not allowed" issue is not  handled on model level, so it is not an error that the test passes anyways, right?

Reply:
Yes that's correct test pass because the validation is not on model level but on view level and on admin page the renaming is avoided due to the usage of read only field.